### PR TITLE
5.2.3 update

### PR DESCRIPTION
**Pull request by g-laures**
Just bumping the version.

I simply fixed 4/5 conflicts.

### CODE REVIEW

**Comment by fpapadopou** [comment]
I see that we already migrated to 5.3.0-snapshot in [this PR](https://github.com/LinearityGmbH/purchases-ios/pull/34). Is this a downgrade or did I miss sth?

**Comment by g-laures** [comment]
I was also a bit surprised, but I think it's just a typo from @myeyesareblind ? The latests commits from the RC team on the other branch suggest that it was version 5.0.0 not 5.3.0

File: RevenueCatUI/Modifiers/ViewExtensions.swift [hunk 1]
```diff
@@ -84,11 +84,20 @@ extension View {
...
+        let iOSVersionWithCrash = OperatingSystemVersion(majorVersion: 18, minorVersion: 0, patchVersion: 0)
+        return ProcessInfo.processInfo.isOperatingSystemAtLeast(iOSVersionWithCrash)
+    }
+
     @ViewBuilder
     // @PublicForExternalTesting
     func scrollableIfNecessary(_ axis: Axis = .vertical, enabled: Bool = true) -> some View {
         if enabled {
-            if #available(iOS 16.0, macCatalyst 16.6, macOS 13.0, tvOS 16.0, watchOS 9.0, *) {
+            if #available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *), !Self.isIOSVersionWithCrash {
```
Comment (myeyesareblind) [inline RevenueCatUI/Modifiers/ViewExtensions.swift]: catalyst-version should be there, and below as well. It's crashing on some macos versions ...

File: RevenueCatUI/UIKit/PaywallViewController.swift [hunk 2]
```diff
@@ -145,13 +145,14 @@ public class PaywallViewController: UIViewController {
...
         self.dismissRequestedHandler = dismissRequestedHandler
+        let handler = PurchaseHandler.default()
 
         self.configuration = .init(
             content: content,
             mode: Self.mode,
             fonts: fonts,
             displayCloseButton: displayCloseButton,
-            purchaseHandler: purchaseHandler
+            purchaseHandler: handler
```
Comment (myeyesareblind) [inline RevenueCatUI/UIKit/PaywallViewController.swift]: whoa, this is not correct

File: RevenueCatUI/PaywallView.swift [hunk 3]
```diff
@@ -103,18 +110,53 @@ public struct PaywallView: View {
 
     // @PublicForExternalTesting
     init(configuration: PaywallViewConfiguration) {
+        let purchaseHandler = configuration.purchaseHandler ?? .default()
+        self._purchaseHandler = .init(wrappedValue: purchaseHandler)
```
Comment (myeyesareblind) [inline RevenueCatUI/PaywallView.swift:114]: it's a bit hard to follow, are we still calling into 'refreshSubscriptions'?

**Comment by myeyesareblind** [comment]
`MyAppPurchaseLogic` looks like they have added some new API that might help us get rid of 'refreshSubscription' altogether

**Comment by myeyesareblind** [comment]
> I was also a bit surprised, but I think it's just a typo from @myeyesareblind ? The latests commits from the RC team on the other branch suggest that it was version 5.0.0 not 5.3.0

looks like typo indeed, where did I get 5.3.0 from at all? 🤷

File: RevenueCatUI/Modifiers/ViewExtensions.swift (hunk 1 above)
Comment (g-laures) [reply RevenueCatUI/Modifiers/ViewExtensions.swift]: Not sure why they changed it.

File: RevenueCatUI/PaywallView.swift (hunk 3 above)
Comment (g-laures) [reply RevenueCatUI/PaywallView.swift:114]: They do this because of `checkForConfigurationConsistency` bellow. But I don't understand your question.

File: RevenueCatUI/Modifiers/ViewExtensions.swift (hunk 1 above)
Comment (myeyesareblind) [reply RevenueCatUI/Modifiers/ViewExtensions.swift]: There are 2 changes here (compared to parent)
1. `macCatalyst 16.6` - this was added by me (you can lookup PRs here)
2. Self.isIOSVersionWithCrash - this is their recent addition

We should take both conditions

File: RevenueCatUI/PaywallView.swift (hunk 3 above)
Comment (myeyesareblind) [reply RevenueCatUI/PaywallView.swift:114]: We've patched RC SDK to add `refreshSubscriptions` to PurchaseHandler. This function runs just after user makes a purchase and is injected from Curve/Move. It looks like it won't run anymore.

File: RevenueCatUI/UIKit/PaywallViewController.swift (hunk 2 above)
Comment (g-laures) [reply RevenueCatUI/UIKit/PaywallViewController.swift]: Oh yeah, sorry, I missed the part above. I merged the two behaviors.

File: RevenueCatUI/PaywallView.swift (hunk 3 above)
Comment (g-laures) [reply RevenueCatUI/PaywallView.swift:114]: I don't think it changes anything as the `purchaseHandler` is initiated the same way, there is just this additional temporary value.

**Review by myeyesareblind — APPROVED** [review]
